### PR TITLE
Global SmartStore functionality

### PR DESCRIPTION
**Pull request by khawkins**

### CODE REVIEW

**Comment by khawkins** [comment]
Hold off for a sec.  I've got a couple of clean-up changes I want to make here.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreDatabaseManager.h [hunk 1]
```diff
@@ -46,6 +46,8 @@ extern NSString * const kSFSmartStoreDbErrorDomain;
  */
 + (SFSmartStoreDatabaseManager *)sharedManagerForUser:(SFUserAccount *)user;
 
++ (SFSmartStoreDatabaseManager *)sharedGlobalManager;
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreDatabaseManager.h]: Shouldn't this be ```(SFSmartStoreDatabaseManager *)sharedGlobalManager:(NSString*)dbName;```? Will the same manager be used across multiple global stores?

Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SmartStore/SFSmartStoreDatabaseManager.h]: Yes.  The database manager (for users too) doesn't work with a specific database—it works with all databases for a given context.

**Comment by khawkins** [comment]
(I'm finished with updates now as well.  Feel free to review.)

**Comment by bhariharan** [comment]
Looks good to me. :+1:

**Pull request merged**